corehq/apps/geospatial/reports.py

```diff
@@ -50,11 +51,14 @@ def headers(self):
         headers.custom_sort = [[2, 'desc']]
         return headers
 
+    def _get_geo_point(self, case):
+        return NotImplementedError()
+
     @property
     def rows(self):
 
         def _get_geo_location(case):
-            geo_point = case.get(get_geo_case_property(case.get('domain')))
+            geo_point = self._get_geo_point(case)
```

Comment:
If you pulled `_get_geo_location()` out of the `rows()` property and made it a static method, then you could write a test for it. :smirk:

Reply:
(Actually, never mind, I'm working in this area, and will probably move this around a bit.)

Reply:
@zandre-eng i am confused though why the function was extracted but it's definition was `NotImplementedError()`
It does look like its definition won't change for classes that inherit it.

I see later that there is a call `case_obj = wrap_case_search_hit(case)` to use this method. I think ideally this function could simply stay in base class and expect a case object always.

Reply:
@mkangia As per our offline discussion, I have opted to make the `CaseManagementMap` class use `CaseSearchES` as well. Now, since all the reports are utiizing the same ES class, we don't have to have `_get_geo_location()` anymore and can simply treat each row the same in `rows`.

Reply:
Great :+1: